x/bank/keeper/send.go

```diff
@@ -24,8 +24,11 @@ type SendKeeper interface {
 	SetBalance(ctx sdk.Context, addr sdk.AccAddress, balance sdk.Coin) error
 	SetBalances(ctx sdk.Context, addr sdk.AccAddress, balances sdk.Coins) error
 
-	GetSendEnabled(ctx sdk.Context) bool
-	SetSendEnabled(ctx sdk.Context, enabled bool)
+	GetParams(ctx sdk.Context) types.Params
+	SetParams(ctx sdk.Context, params types.Params)
+
+	GetSendEnabled(ctx sdk.Context, denom string) bool
+	CoinsSendEnabled(ctx sdk.Context, coins ...sdk.Coin) error
```

Comment:
```suggestion
	SendEnabledCoins(ctx sdk.Context, coins ...sdk.Coin) error
```

Reply:
On second thought, I'm thinking this should rather take a variadic list of denominations (`...string`) and be called `SendEnabledDenoms` to complement `SendEnabledDenom`.

Reply:
After reviewing the current usages of these methods, in all cases the arguments for `denom` are passed in using `coin.Denom`.  In addition using the `denom` string means that when a structure of `coins` is present the `denom` string must be extracted and passed into these functions (often adding yet another for-range loop).  Based on this review and the existing methods which are all based on `sdk.Coin` it would seem that for consistency we should be using the `Coin` type and not a `denom` string.  Further a `denom` string may not be a valid denom at all when it is handled outside of the `Coin` class.

Reply:
You can assume all denoms are valid, and you can easily implement a method `Coins#Denominations` that returns a unique list, so then it becomes `SendEnabledDenoms(coins.Denoms())`.